clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -58,7 +58,7 @@
      * 
      * @param topic The topic the record will be appended to
      * @param partition The partition to which the record should be sent
-     * @param timestamp The timestamp of the record
+     * @param timestamp The timestamp of the record, in milliseconds since epoch.
```

Comment:
Since we're updating the javadoc, maybe we should mention the effect of using `null`?

Reply:
What is the effect of null? That the server assigns the timestamp, which is the log append time?

Reply:
I think the effect is that the producer will assign the timestamp using `System.currentTime` before sending.